packages/sdk-react/src/components/receivables/InvoiceDetails/ExistingInvoiceDetails/components/EmailInvoiceDetails.tsx

```diff
@@ -59,30 +62,95 @@ interface EmailInvoiceDetailsProps {
   invoiceId: string;
   onClose: () => void;
 }
+interface EmailInvoiceFormProps extends EmailInvoiceDetailsProps {
+  subject: string;
+  body: string;
+  to: string;
+  isLoading: boolean;
+}
+
+export const EmailInvoiceDetails = (props: EmailInvoiceDetailsProps) => {
+  const { i18n } = useLingui();
+  const { data: me, isLoading: isLoadingUser } = useMe();
+  const { data: receivable, isLoading: isLoadingReceivable } =
+    useReceivableById(props.invoiceId);
+  const { data: contacts, isLoading: isLoadingContacts } =
+    useReceivableContacts(props.invoiceId);
+  const { entityName, isLoading: isLoadingEntity } = useMyEntity();
+
+  const defaultContact =
+    contacts && contacts.length
+      ? contacts?.find((c) => c.is_default) || contacts?.[0]
+      : undefined;
+
+  const to = defaultContact?.email ?? '';
+
+  const body =
+    defaultContact && me
+      ? t(i18n)`Hi ${defaultContact.first_name},
 
-export const EmailInvoiceDetails = (props: EmailInvoiceDetailsProps) => (
-  <MoniteScopedProviders>
-    <EmailInvoiceDetailsBase {...props} />
```

Comment:
what is the reason to split logic into `EmailInvoiceDetails` & `EmailInvoiceDetailsBase`?
Usually, the pattern is as follows:
we put everything the component needs into the `ComponentBase`, then wrap it with `MoniteScopedProviders` and import as the `Component`.
Moreover, we don't have to wrap those components with `MoniteScopedProviders` that we don't export from the SDK. Could you consider removing this wrapper? Probably one of its parents already wrapped with it.

Reply:
Thanks for that context. The split (`EmailInvoiceDetails` & `EmailInvoiceDetailsBase`) and `MoniteScopedProviders` are part of already existing code before the cherry-picking, but I am happy to make further modifications.

I would want to keep the pattern of passing down the default form values as props for the `Form` compared to the previous approach as it extends reusability for other use cases; it's also a recommended approach for the react-hook-form library for resetting form values, improves performance, and reduces the number of re-renders and potential of unexpected side effects from the checks in the previous implementation.

Reply:
ok, let's leave it then.
I agree with approach of passing down the default form values